fairworkflows/fairstep.py

```diff
@@ -13,6 +14,39 @@
                        namespaces.PPLAN.hasOutputVar, DCTERMS.description, RDFS.label]
 
 
+class FairVariable:
+    """Represents a variable.
+
+    The variable corresponds to a blank node that has 2 RDF:type relations: (1) PPLAN:Variable,
+    and (2) a string literal representing the type (i.e. int, str, float) of the variable.
+
+    The FairVariable is normally associated with a FairStep by a PPLAN:hasInputVar or
+    PPLAN:hasOutputVar predicate.
+
+    Attributes:
+        name: The name of the variable (and of the blank node in rdf)
+        uri: The uri that the variable is referred to (usually only set when we extract a
```

Comment:
> usually only set when we extract a variable from rdf)

I'm a bit more confused by this comment. Would this mean the URI is not fixed (static) in the published RDF? So every time a step is loaded it gets a different URI for that instance?

Reply:
I can see that the otherway around, when publishing a new step, it converts the BNode to some URI.

Reply:
OK, yes this is confusing. I made the documentation a lot clearer now.